subworkflows/local/input_check.nf

```diff
@@ -66,13 +66,75 @@ workflow INPUT_CHECK {
         ch_raw_long_reads = Channel.empty()
     }
 
+    if (params.assembly_input) {
+        // check if
+        if(!hasExtension(params.input, "csv")) { exit 1, "ERROR: when supplying assemblies with --assembly_input, reads must be supplied using a CSV!"}
+
+        ch_input_assembly_rows = Channel
+            .from(file(params.assembly_input))
+            .splitCsv(header: true)
+            .map { row ->
+                    if (row.size() == 3) {
+                        def id        = row.id
+                        def assembler = row.assembler ?:  false
+                        def assembly  = row.fasta ? file(row.fasta, checkIfExists: true) : false
+                        // Check if given combination is valid
+                        if (!assembly) exit 1, "Invalid input assembly samplesheet: fasta can not be empty."
+                        if (!assembler) exit 1, "Invalid input assembly samplesheet: assembler can not be empty."
+                        return [ id, assembler, assembly ]
+                    } else {
+                        exit 1, "Input samplesheet contains row with ${row.size()} column(s). Expects 3."
+                    }
+                }
```

Comment:
it seems to me some indents are not right, e.g. line 72 might need one more, line 88 one less?

Reply:
Had another look, looks correct to me - happy to be proven wrong though!